controllers/serviceinstance_controller.go

```diff
@@ -201,16 +201,17 @@ func (r *ServiceInstanceReconciler) createInstance(ctx context.Context, smClient
 		return r.markAsNonTransientError(ctx, smTypes.CREATE, provisionErr, serviceInstance, log)
 	}
 
+	var tags []string
 	if provision.Location != "" {
 		serviceInstance.Status.InstanceID = provision.InstanceID
 		if len(provision.Tags) > 0 {
-			tags, err := getTags(provision.Tags)
+			tags, err = getTags(provision.Tags)
 			if err != nil {
 				log.Error(err, "failed to unmarshal tags")
-			} else {
-				serviceInstance.Status.Tags = tags
 			}
 		}
+		updateInstanceTags(serviceInstance, tags, serviceInstance.Spec.CustomTags)
+
```

Comment:
good point, but on the other hand update of a `ServiceInstance` object will not trigger reconciliation loop for a `ServiceBinding` object, so the new tags won't be updated in the binding's secret

Reply:
Nothing will be updated in the secret once it created because binding is not upatable.
It's not expected that binding  secret will point to new plan if I'll update service plan in ServiceInstance but from now on the bindings will be created accordingly.
So we good here.